src/cobra/flux_analysis/reaction.py

```diff
@@ -138,82 +142,113 @@ def assess_component(model, reaction, side, flux_coefficient_cutoff=0.001, solve
+    """Perform quick optimization and return the objective value.
+
+    The objective value is returned at `value` error value.
+
+    Parameters
+    ----------
+    model: cobra.Model
+        The model to optimize.
+    value: float
+        The error value to consider.
+
+    Returns
+    -------
+    float
+        The optimized value of the model's objective.
+
+    """
     return model.slim_optimize(error_value=value)
 
 
-def assess_precursors(model, reaction, flux_coefficient_cutoff=0.001, solver=None):
-    """Assesses the ability of the model to provide sufficient precursors for
-    a reaction operating at, or beyond, the specified cutoff.
+def assess_precursors(
+    model: "Model",
+    reaction: Reaction,
+    flux_coefficient_cutoff: float = 0.001,
+    solver: str = None,
+) -> Union[bool, Dict]:
+    """Assess production capacity of precursors.
+
+    Assess the ability of the model to provide sufficient precursors for
+    a reaction operating at, or beyond, the `flux_coefficient_cutoff`.
 
-    Deprecated: use assess_component instead
+    .. deprecated:: 0.7.0
+              `assess_precursors` will be removed in cobrapy 1.0.0, it is
+              replaced by `assess_component`.
 
     Parameters
     ----------
     model : cobra.Model
-        The cobra model to assess production capacity for
-
-    reaction : reaction identifier or cobra.Reaction
-        The reaction to assess
-
-    flux_coefficient_cutoff :  float
-        The minimum flux that reaction must carry to be considered active.
-
-    solver : basestring
-        Solver name. If None, the default solver will be used.
+        The cobra model to assess production capacity for.
+    reaction : str or cobra.Reaction
+        The reaction to assess.
+    flux_coefficient_cutoff : float, optional
+        The minimum flux that reaction must carry to be considered active
+        (default 0.001).
+    solver : str, optional
+        The solver name. If None, the default solver will be used
+        (default None).
 
     Returns
     -------
     bool or dict
         True if the precursors can be simultaneously produced at the
         specified cutoff. False, if the model has the capacity to produce
-        each individual precursor at the specified threshold  but not all
-        precursors at the required level simultaneously. Otherwise a
-        dictionary of the required and the produced fluxes for each reactant
-        that is not produced in sufficient quantities.
+        each individual precursor at the specified threshold but not all
+        precursors at the required level simultaneously. Otherwise, a
+        dictionary of the required and the produced fluxes for each
+        reactant that is not produced in sufficient quantities.
 
     """
-    warn("use assess_component instead", DeprecationWarning)
+    warn("Use cobra.sampling.reaction.assess_component() instead.", DeprecationWarning)
```

Comment:
I still learn things about the codebase 🙂 It's actually a bit odd to me that `assess_component` is part of the sampling sub-package.

Reply:
Yeah that is surprising. I think once the deprecations are executed, there will be a better understanding of how to restructure the project better.